### ISSUE TITLE
Disable Cashshop When Talk to NPC

### ISSUE THREAD

**Issue by mazvi** [rathena/rathena#7168]
**Is your request related to a missing feature?** No

**Describe the solution you'd like** Disable Cashshop When Talk to NPC

First talk to the NPC, then open the cashshop and shop it. After that, continue the NPC that reduces the cashshop, this is a potential bug.

**Describe alternatives you've considered**

**Additional context**

**Comment by gidzdlcrz** [comment]
I dont understand the problem. So you were saying when you are talking with npc then you decided to open the cash shop and buy something?

Well in my test i can talk to npc and at the same time i can buy stuff in my cash shop. There is no problem at all.

**Comment by mazvi** [comment]
> I dont understand the problem. So you were saying when you are talking with npc then you decided to open the cash shop and buy something?
> 
> Well in my test i can talk to npc and at the same time i can buy stuff in my cash shop. There is no problem at all.

yes, talk NPC with have script eg: #CASHPOINTS -= 10000 on last script

after talk with that NPC do not next on that NPC, but buy on cashshop on right top until you dont have 10000

then next until script #CASHPOINTS -= 10000 on last script

this will bug

**Comment by gidzdlcrz** [comment]
> > I dont understand the problem. So you were saying when you are talking with npc then you decided to open the cash shop and buy something?
> > Well in my test i can talk to npc and at the same time i can buy stuff in my cash shop. There is no problem at all.
> 
> yes, talk NPC with have script eg: #CASHPOINTS -= 10000 on last script
> 
> after talk with that NPC do not next on that NPC, but buy on cashshop on right top until you dont have 10000
> 
> then next until script #CASHPOINTS -= 10000 on last script
> 
> this will bug

Oh okay. By default you cannot use @cash when the cash shop is open. 

```
	// Since there is no cashpoint update packet we need to force updating like this
	if( sd->state.cashshop_open ){
		clif_displaymessage(fd, msg_txt(sd, 1376)); // Please close the cashshop before using this command.
		return -1;
	}
```

This portion is need to be convened into npc script and add to that NPC script of yours.

**Comment by Akkarinage** [comment]
Can you provide the script you're using to reproduce this? I don't know of any use-cases where anyone would keep reducing the `#CASHPOINTS` variable over and over in a script with NPC dialogue.

Personally, this either sounds like shoddy scripting, or I'm not understanding the issue..
